src/kvstore.c

```diff
@@ -378,7 +387,7 @@ size_t kvstoreMemUsage(kvstore *kvs) {
     mem += listLength(kvs->rehashing) * sizeof(listNode);
 
     if (kvs->dict_size_index)
-        mem += sizeof(unsigned long long) * (kvs->num_dicts + 1);
+        mem += sizeof(unsigned long long) * (kvs->allocated_dicts + 1);
```

Comment:
In cluster mode, `num_dicts` is always the size of the slot. Here, the size of the allocated dict should be used.